src/vm/siginfo.cpp

```diff
@@ -4953,11 +4953,54 @@ void PromoteCarefully(promote_func   fn,
     (*fn) (ppObj, sc, flags);
 }
 
+void ReportByRefPointersFromByRefLikeObject(promote_func *fn, ScanContext *sc, PTR_MethodTable pMT, PTR_VOID pSrc)
+{
+    WRAPPER_NO_CONTRACT;
+
+    _ASSERTE(pMT->IsByRefLike());
+
+    if (pMT == g_TypedReferenceMT)
```

Comment:
The first field of TypedReference is actually byref, but it is not recorded as byref in metadata. It has been special cased for GC reporting in multiple places. This is yet another places that should be special cased for it.

Ideally, we would change the implementation of TypedReference to:

```
struct TypedReference
{
    ByReference<byte> _value;
    IntPtr _type;
...
}
```

and get rid of the special casing. But until/if that happens, we should continue special casing it.